Added lots of examples and links to babel features and stages

adriantoine opened the pull request:
Hi!

As @hzoo asked me here: https://github.com/adriantoine/inside-babel-stages/issues/1, I've added examples and explanations for lots of babel features to the babel docs as it was done on my repo https://github.com/adriantoine/inside-babel-stages.

Please have a look, here are all the changed files:
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/syntax-trailing-function-commas.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-class-constructor-call.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-class-properties.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-decorators.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-do-expressions.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-exponentiation-operator.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-export-extensions.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-function-bind.md
- https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-object-rest-spread.md

English is not my native language, so, feel free to correct me! :stuck_out_tongue: 

Note: I added a "Try in REPL" link to most of examples but it won't work on the github preview because I've used an absolute path `/repl/...`, but it will work on the babel docs website.

hzoo commented:
Awesome which check it out soon. 🎉

hzoo commented on docs/plugins/syntax-trailing-function-commas.md:
```diff
@@ -9,6 +9,83 @@ package: babel-plugin-syntax-trailing-function-commas
+
+## Example
+### Basic
+This is an example from the Proposal.
```
You could link to the Proposal here

hzoo commented on docs/plugins/syntax-trailing-function-commas.md:
````diff
@@ -9,6 +9,83 @@ package: babel-plugin-syntax-trailing-function-commas
+
+You want to have a new parameter called `param3`:
+
+```js
````
you could also try a `diff` syntax here

``` `diff` ```

```diff
function clownPuppiesEverywhere(
  param1,
+  param2, // Change this line to add a comma
+  param3  // Add param3
) { /* ... */ }
```

adriantoine commented:
@hzoo I've updated the PR with your remarks: https://github.com/adriantoine/babel.github.io/commit/75304276a0be0c1686dde7445b6d73c9fd4910c5, I also added a link to this great article: https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8 which explains the benefits of the trailing comma approach very nicely.

hzoo commented on docs/plugins/transform-class-constructor-call.md:
```diff
@@ -6,6 +6,65 @@ permalink: /docs/plugins/transform-class-constructor-call/
 package: babel-plugin-transform-class-constructor-call
 ---
 
+Allows to call a class constructor
```
Actually I think we want to put a deprecation message here since the proposal was dropped. https://github.com/babel/babel/wiki/Babel-7

adriantoine replied:
Sure, actually I didn't know :stuck_out_tongue: , I will add it

hzoo commented on docs/plugins/transform-class-properties.md:
```diff
@@ -6,6 +6,93 @@ permalink: /docs/plugins/transform-class-properties/
 package: babel-plugin-transform-class-properties
 ---
 
+Allows to use class properties
```
Maybe change all of the headings to say something like: `This plugin allows Babel to transform x`

hzoo commented on docs/plugins/syntax-trailing-function-commas.md:
````diff
@@ -9,6 +9,85 @@ package: babel-plugin-syntax-trailing-function-commas
 This plugin allows Babel to parse [trailing function commas](https://github.com/jeffmo/es-trailing-function-commas).
 
+```js
+function clownPuppiesEverywhere(
````
Might not be too useful but maybe we can add a repl link here just to be consistent?

hzoo commented on docs/plugins/transform-class-properties.md:
````diff
@@ -6,6 +6,93 @@ permalink: /docs/plugins/transform-class-properties/
+```
+
+### React propTypes
+According to the [React documentation](https://facebook.github.io/react/docs/reusable-components.html#es6-classes), the way to add `propTypes` to a React ES6 class component is like that:
````
Not sure if we want the react example - could link to the docs instead

adriantoine commented:
Quick update:
- I've added the REPL link to [syntax-trailing-function-commas](https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/syntax-trailing-function-commas.md)
- Also added a deprecation warning to [transform-class-constructor-call](https://github.com/adriantoine/babel.github.io/blob/master/docs/plugins/transform-class-constructor-call.md)

Now I need to add a more descriptive header `This plugin allows Babel to transform ...` to transforms.

adriantoine commented:
Ok so now [I changed most of the transform headers](https://github.com/babel/babel.github.io/pull/795/commits/a11a39d2dd4ec00846f4b1e1d1881253b98e97a0). I didn't change the es2015 ones (should I?) or the ones I didn't clearly understand.

@hzoo can you have a look again? Thanks!

hzoo commented:
Sorry for the long wait - I reviewed most of it and it was kinda long and never got back to this.. thanks again.

adriantoine commented:
No problem thanks @hzoo! 😄
